flight/PiOS/inc/pios_lis3mdl.h

```diff
@@ -0,0 +1,64 @@
+/**
+ ******************************************************************************
+ * @file       pios_lis3mdl.h
+ * @author     dRonin, http://dRonin.org/, Copyright (C) 2016
+ * @addtogroup PIOS PIOS Core hardware abstraction layer
+ * @{
+ * @addtogroup PIOS_LIS3MDL
+ * @{
+ * @brief Driver for ST LIS3MDL Magnetometer
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ * Additional note on redistribution: The copyright and license notices above
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+#ifndef PIOS_LIS3MDL_H
+#define PIOS_LIS3MDL_H
+
+#include "pios.h"
+
+enum pios_lis3mdl_orientation { // clockwise rotation from board forward
+	PIOS_LIS_TOP_0DEG    = 0x00,
+	PIOS_LIS_TOP_90DEG   = 0x01,
+	PIOS_LIS_TOP_180DEG  = 0x02,
+	PIOS_LIS_TOP_270DEG  = 0x03,
+	PIOS_LIS_BOTTOM_0DEG  = 0x04,
+	PIOS_LIS_BOTTOM_90DEG  = 0x05,
+	PIOS_LIS_BOTTOM_180DEG  = 0x06,
+	PIOS_LIS_BOTTOM_270DEG  = 0x07,
+};
+
+struct pios_lis3mdl_cfg {
+	enum pios_lis3mdl_orientation orientation;
+};
+
+typedef struct lis3mdl_dev * lis3mdl_dev_t;
```

Comment:
Yep, let's agree to do something about defining our own rules rather than sticking with Kernel rules. They are good but also have lots of irrelevant stuff.